ocaml/xapi/xapi.ml

```diff
@@ -550,14 +551,27 @@ let check_network_reset () =
          (* Introduce PIFs for remaining interfaces *)
          Xapi_pif.scan ~__context ~host;
 
+         (* Create a vlan PIF if management interface asked on a VLAN *)
+         let create_vlan pif vlan =
+           let name_label = Printf.sprintf "Pool-wide network associated with %s on VLAN%s" device vlan in
+           let network = Xapi_network.create ~__context ~name_label ~name_description:"" ~mTU:1500L ~other_config:[] ~tags:[] in
+           let vlan = Xapi_vlan.create ~__context ~tagged_PIF:pif ~network ~tag:(Int64.of_string vlan) in
+           Db.VLAN.get_untagged_PIF ~__context ~self:vlan
+         in
+
          (* Introduce and configure the management PIF *)
          let pifs = Db.PIF.get_refs_where ~__context ~expr:(And (
              Eq (Field "host", Literal (Ref.string_of host)),
              Eq (Field "device", Literal device)
            )) in
          match pifs with
-         | [] -> error "management PIF %s not found" device
-         | pif :: _ ->
+         | []           -> error "management PIF %s not found" device
```

Comment:
I still think that a flat match over `pifs` and `vlan` at the same time is more readable but I'm fine with this, too.